src/process.c

```diff
@@ -113,6 +113,115 @@ char * rcutils_get_executable_name(rcutils_allocator_t allocator)
   return executable_name;
 }
 
+#if defined _WIN32 || defined __CYGWIN__
+static
+rcutils_ret_t
+append_backslashes(rcutils_char_array_t * char_array, size_t n)
+{
+  if (n <= 0) {
+    return RCUTILS_RET_OK;
+  }
+
+  size_t current_strlen;
+  if (char_array->buffer_length == 0) {
+    current_strlen = 0;
+  } else {
+    // The buffer length always contains the trailing \0, so the strlen is one less than that.
+    current_strlen = char_array->buffer_length - 1;
+  }
+
+  size_t new_length = current_strlen + n + 1;
+  rcutils_ret_t ret = rcutils_char_array_expand_as_needed(char_array, new_length);
+  if (RCUTILS_RET_OK != ret) {
+    return ret;
+  }
+
+  memset(char_array->buffer + current_strlen, '\\', n);
+  char_array->buffer[new_length - 1] = '\0';
+
+  char_array->buffer_length = new_length;
+
```

Comment:
This store is redundant? i believe that line 134 above, `rcutils_char_array_expand_as_needed` call also uddates the buffer_length with new_length.

```suggestion
```

Reply:
It wouldn't change the length, no. The call to `_expand_as_needed` will update the capacity if necessary, but the length only gets modified if you shrink the array.